src/project.rs

```diff
@@ -92,8 +95,21 @@ pub fn analysed(inputs: Vec<Input>) -> Result<Vec<Analysed>, Error> {
 
         println!("Compiling {}", name_string);
 
-        let ast = crate::typ::infer_module(module, &modules_type_infos)
-            .map_err(|error| Error::Type { path, src, error })?;
+        let mut warnings = Vec::new();
+        let (result, mut module_warnings) = crate::typ::infer_module(module, &modules_type_infos);
+        module_warnings.drain(..).for_each(|warning| {
```

Comment:
Could this be `module_warnings.map` ? Would be slightly more concise and the vector wouldn't be grown as new warnings are added.

Reply:
Yeah, probably. `.map` was being a pain when it was part of the module struct but might be ok now. Will give it a go.

Reply:
Thanks

Reply:
Yeah that works fine, I was just being dumb earlier. Updated.

Reply:
I'm sure you wern't :)